src/lib/storage/run_length_column.cpp

```diff
@@ -0,0 +1,73 @@
+#include "run_length_column.hpp"
+
+#include <algorithm>
+
+#include "storage/value_column.hpp"
+#include "type_cast.hpp"
+#include "utils/assert.hpp"
+#include "utils/performance_warning.hpp"
+
+namespace opossum {
+
+template <typename T>
+RunLengthColumn<T>::RunLengthColumn(const std::shared_ptr<const pmr_vector<T>>& values,
+                                    const std::shared_ptr<const pmr_vector<bool>>& null_values,
+                                    const std::shared_ptr<const pmr_vector<ChunkOffset>>& end_positions)
+    : _values{values}, _null_values{null_values}, _end_positions{end_positions} {}
+
+template <typename T>
+std::shared_ptr<const pmr_vector<T>> RunLengthColumn<T>::values() const {
+  return _values;
+}
+
+template <typename T>
+std::shared_ptr<const pmr_vector<bool>> RunLengthColumn<T>::null_values() const {
+  return _null_values;
+}
+
+template <typename T>
+std::shared_ptr<const pmr_vector<ChunkOffset>> RunLengthColumn<T>::end_positions() const {
+  return _end_positions;
+}
+
+template <typename T>
+const AllTypeVariant RunLengthColumn<T>::operator[](const ChunkOffset chunk_offset) const {
+  PerformanceWarning("operator[] used");
+
+  const auto end_position_it = std::lower_bound(_end_positions->cbegin(), _end_positions->cend(), chunk_offset);
+  const auto index = std::distance(_end_positions->cbegin(), end_position_it);
+
+  const auto value = (*_values)[index];
+  const auto is_null = (*_null_values)[index];
+
+  if (is_null) return NULL_VALUE;
+
+  return AllTypeVariant{value};
+}
+
+template <typename T>
+size_t RunLengthColumn<T>::size() const {
+  DebugAssert(_end_positions->size() > 0u, "Column size is never zero.");
```

Comment:
why not?

Reply:
When would that make sense? An empty run-length encoded column?

Reply:
I removed it and it returns `0u` now in case it’s empty.

Reply:
It does not make sense. It is, however, not the data structure's job to decide if this makes sense or not.